command/agent_test.go

```diff
@@ -2612,14 +2614,13 @@ listener "tcp" {
 	cmd.client = serverClient
 	cmd.startedCh = make(chan struct{})
 
+	var output string
 	wg := &sync.WaitGroup{}
 	wg.Add(1)
 	go func() {
 		code := cmd.Run([]string{"-config", configPath})
 		if code != 0 {
-			t.Errorf("non-zero return code when running agent: %d", code)
```

Comment:
I don't quite see how this line could be triggering an issue, since the `t.Errorf` is _before_ a `wg.Done()`. Is this line causing an issue? I'd be interested to see the stacktrace!

Reply:
Sample stack trace could be found [here](https://github.com/hashicorp/vault-enterprise/actions/runs/6227498697/job/16902424827)

It is not on the exact same line, but in a similar code construct.
https://github.com/hashicorp/vault/blob/e48a57c5ce84355f0d886cb8fc85241d388d17c3/command/agent_test.go#L2918

```
panic: Fail in goroutine after TestAgent_Config_ReloadTls has completed

goroutine 212277 [running]:
testing.(*common).Fail(0xc00a4cd1e0)
	/home/runner/actions-runner/_work/_tool/go/1.21.1/x64/src/testing/testing.go:952 +0xd4
testing.(*common).Errorf(0xc00a4cd1e0, {0xa9c2b7a?, 0x121?}, {0xc008bebfc0?, 0xc00437cf60?, 0x3e0cb25?})
	/home/runner/actions-runner/_work/_tool/go/1.21.1/x64/src/testing/testing.go:1069 +0x5e
github.com/hashicorp/vault/command.TestAgent_Config_ReloadTls.func1()
	/home/runner/actions-runner/_work/vault-enterprise/vault-enterprise/command/agent_test.go:2918 +0x115
created by github.com/hashicorp/vault/command.TestAgent_Config_ReloadTls in goroutine 212274
	/home/runner/actions-runner/_work/vault-enterprise/vault-enterprise/command/agent_test.go:2915 +0xa1f
```

Reply:
Thanks! I appreciate it